nilearn/glm/first_level/first_level.py

```diff
@@ -513,6 +514,26 @@ def _check_fit_inputs(
         if not isinstance(run_imgs, (list, tuple)):
             run_imgs = [run_imgs]
 
+        # check that mask type and image types are compatible
+        volumetric_type = (Nifti1Image, NiftiMasker, str, Path)
+        surface_type = (SurfaceImage, SurfaceMasker)
+        if self.mask_img is not None:
+            if (
+                isinstance(self.mask_img, volumetric_type)
+                and any(
+                    not isinstance(x, (Nifti1Image, str, Path))
+                    for x in run_imgs
+                )
+            ) or (
+                isinstance(self.mask_img, surface_type)
+                and any(not isinstance(x, SurfaceImage) for x in run_imgs)
+            ):
+                raise TypeError(
+                    "Mask and images to fit must be of the same type.\n",
```

Comment:
Not sure if the error message is super clear.
Should it be more specific?
For example: if mask is Surface image or Surface masker, then you must pass SurafaceImage objects to fit.